walletnode/services/artworkregister/task.go

```diff
@@ -198,6 +213,62 @@ func (task *Task) run(ctx context.Context) error {
 	return nil
 }
 
+// generateRedundantIDs generates redundant IDs and assigns to task.redundantIDs
+func (task *Task) generateRedundantIDs() error {
+	ddDataJson, err := json.Marshal(task.fingerprintAndScores)
+	if err != nil {
+		return errors.Errorf("failed to marshal dd-data: %w", err)
+	}
+
+	ddEncoded := base64.RawURLEncoding.EncodeToString(ddDataJson)
+	signSN1 := base64.RawURLEncoding.EncodeToString(task.signatures[0])
+	signSN2 := base64.RawURLEncoding.EncodeToString(task.signatures[1])
+	signSN3 := base64.RawURLEncoding.EncodeToString(task.signatures[2])
+
+	const digitBytes = "0123456789"
+	b := make([]byte, 4)
+	for i := range b {
+		b[i] = digitBytes[rand.Intn(len(digitBytes))]
+	}
+	randNumStr := string(b)
+
+	randNum, err := strconv.Atoi(randNumStr)
+	if err != nil {
+		return errors.Errorf("invalid random 4 bytes number: %w", err)
```

Comment:
I think randUint32() is enough for 4 bytes with higher range.
And the error due to your random generator should not pause the registration.